api/src/org/apache/cloudstack/api/command/user/vm/DeployVMCmd.java

```diff
@@ -192,6 +194,10 @@
             + " Example: dhcpoptionsnetworklist[0].dhcp:114=url&dhcpoptionsetworklist[0].networkid=networkid&dhcpoptionsetworklist[0].dhcp:66=www.test.com")
     private Map dhcpOptionsNetworkList;
 
+    @Parameter(name = ApiConstants.DATADISK_OFFERING_LIST, type = CommandType.MAP, since = "4.10", description = "datadisk template to disk-offering mapping;" +
+            " an optional parameter used to create additional data disks from datadisk templates; can't be specified with diskOfferingId parameter")
+    private Map dataDiskTemplateToDiskOfferingList;
```

Comment:
Come on! Do not give a name ending in list for an attribute that is a map. It is misleading.

What is the structure of this attribute? A map of maps?!

Reply:
It is a map of maps indeed. I agree that its name is not the best, however I noticed the same situation on previous parameter dhcpOptionsNetworkList. I think this PR's author tried to follow that pattern. I hadn't changed its name yet, which name do you suggest for it?

Reply:
It is always time to create new patterns :)

I would suggest using a qualifying name that describes the true object. Also, a documentation; for instance, this is a map of maps. What is the first key that maps an entry to a map? What is the second key that maps a value?

After having this information we can even question, do we need a map of maps structure? It feels a pretty complicated solution....